include/vcpkg/base/optional.h

```diff
@@ -253,6 +253,12 @@ namespace vcpkg
 
         constexpr explicit operator bool() const { return this->m_base.has_value(); }
 
+        T& emplace()
```

Comment:
Consider also adding a ctor that does this.

Reply:
Not sure precisely what such a ctor would look like (other than some magic "default construct it please" tag)

Reply:
Exactly that, so that the construction can happen directly in the ctor (and follow normal EH rules for that etc.)